### ISSUE TITLE
Intermittent precompilation failure

### ISSUE THREAD

**Issue by Cody-G** [timholy/ImagineFormat.jl#25]
I'm seeing the below failure intermittently on my laptop and consistently with CI:

```julia
ERROR: LoadError: syntax: extra token "Endian" after end of expression
Stacktrace:
 [1] include at .\boot.jl:317 [inlined]
 [2] include_relative(::Module, ::String) at .\loading.jl:1038
 [3] include(::Module, ::String) at .\sysimg.jl:29
 [4] top-level scope at none:2
 [5] eval at .\boot.jl:319 [inlined]
 [6] eval(::Expr) at .\client.jl:399
 [7] top-level scope at .\none:3
in expression starting at C:\Users\appveyor\.julia\packages\ImagineFormat\JcVxt\src\ImagineFormat.jl:79
ERROR: LoadError: Failed to precompile ImagineFormat [4bab44a2-5ff2-5a6b-8e10-825fb9ac126a] to C:\Users\appveyor\.julia\compiled\v0.7\ImagineFormat\CrjpI.ji.
```

I'm not quite sure why I see the error intermittently locally when I'm not modifying the source in `ImagineFormat.jl`.  Maybe recompilation gets triggered in other scenarios too?  Anyway I'm guessing that precompilation of this module is somehow dangerous, though I haven't figured out why yet.

**Comment by Cody-G** [comment]
I just noticed that this happens on Julia 0.7 but not 1.0.  Also the failure gets triggered when ImagineFormat is getting precompiled as a dependency of another module that uses Reexport.  It doesn't reexport ImagineFormat, but still I wonder if Reexport is somehow involved here.

**Comment by timholy** [comment]
Interesting. There are a huge number of fixes that are in 1.0.3, if that's what you mean by 1.0. (e.g., see https://github.com/JuliaLang/julia/compare/release-0.7...release-1.0.) Not surprising that you'd encounter bugs that are visible only on 0.7.

**Comment by Cody-G** [comment]
Yes it seems this is either a Julia 0.7 issue or a Reexport.jl issue (they seem to have made changes there for the 0.7 -> 1.0 transition) and probably has nothing to do with ImagineFormat so I'm closing.  Thanks!
